### ISSUE TITLE
[API] Add endpoint to delete PRs

### ISSUE THREAD

**Issue by jthvai** [go-gitea/gitea#27841]
### Feature Description

TL;DR:
```
DELETE /repos/{owner}/{repo}/pulls/{index}
```
***
This endpoint should be equivalent to the "Delete" button on a pull request's page (refer to screenshot below).

<details><summary>Screenshot: PR page with the Delete button highlighted</summary>

![example](https://github.com/go-gitea/gitea/assets/49891202/2a62d666-c4c0-411f-a299-96c4d6b5663c)

</details>

An equivalent endpoint doesn't exist in [GitHub's API](https://docs.github.com/en/rest/pulls/pulls), but it would make sense to implement it like [deleting an issue](https://try.gitea.io/api/swagger#/issue/issueDelete) (where a `DELETE` request is sent to the same route as `GET`ing an issue).

**Comment by jthvai** [comment]
Closing as functionality is provided by `DELETE /repos/{owner}/{repo}/issues/{index}`; issues and PRs are the same internally.
